core/src/main/scala/kafka/coordinator/GroupMetadataManager.scala

```diff
@@ -166,10 +168,12 @@ class GroupMetadataManager(val brokerId: Int,
   def prepareStoreGroup(group: GroupMetadata,
                         groupAssignment: Map[String, Array[Byte]],
                         responseCallback: Short => Unit): DelayedStore = {
-    // construct the message to append
+    // construct the message to append. GroupMetadata does not use message timestamp so we use magic value 0
```

Comment:
It's probably useful to use the message format v1 on the topic. This can avoid the recompression overhead when compression is enabled.